client/src/components/Form/Elements/FormData/FormData.vue

```diff
@@ -0,0 +1,521 @@
+            };
+            if (option.keep) {
+                const keepKey = `${option.id}_${option.src}`;
+                keepOptions.value[keepKey] = newOption;
+            } else {
+                const accepted = !props.tag || option.tags.includes(props.tag);
+                if (accepted) {
+                    result.push(newOption);
+                }
+            }
+        });
+        // Populate keep-options from cache
+        Object.entries(keepOptions.value).forEach(([key, option]) => {
+            if (option.value?.src === currentSource.value) {
+                result.unshift(option);
+            }
+        });
+        // Sort entries by hid
+        result.sort((a, b) => {
+            const aHid = a.value && a.value.hid;
+            const bHid = b.value && b.value.hid;
+            if (aHid && bHid) {
+                return bHid - aHid;
+            } else {
+                return 0;
+            }
+        });
+        // Add optional entry
+        if (!currentVariant.value?.multiple && props.optional) {
+            result.unshift({
+                label: "Nothing selected",
+                value: null,
+            });
+        }
+        return result;
+    } else {
+        return [];
+    }
+});
+
+/**
+ * Dataset collection elements should be treated as plain values.
+ */
+const isDCE = computed(() => {
+    if (props.value && props.value.values.length > 0) {
+        return props.value.values.findIndex((v) => v.src !== SOURCE.COLLECTION_ELEMENT) === -1;
+    } else {
+        return false;
+    }
+});
+const isLDDA = computed(() => {
+    if (props.value && props.value.values.length > 0) {
+        return props.value.values.findIndex((v) => v.src !== SOURCE.LIBRARY_DATASET) === -1;
+    } else {
+        return false;
+    }
+});
+/**
+ * Provides placeholder label for select field
+ */
+const placeholder = computed(() => (currentSource.value === SOURCE.DATASET ? "dataset" : "dataset collection"));
+
+/**
+ * Provides the array of available variants associated with a specific form data type
+ */
+const variant = computed(() => {
+    const flavorKey = props.flavor ? `${props.flavor}_` : "";
+    const multipleKey = props.multiple ? `_multiple` : "";
+    const variantKey = `${flavorKey}${props.type}${multipleKey}`;
+    return VARIANTS[variantKey];
+});
+
+/**
+ * Clears highlighting with delay
+ */
+function clearHighlighting(timeout = 1000) {
+    setTimeout(() => {
+        currentHighlighting.value = null;
+    }, timeout);
+}
+
+/** Add values from drag/drop or data dialog sources */
+function handleIncoming(incoming: Record<string, unknown>, partial = true) {
+    if (incoming) {
+        const values = Array.isArray(incoming) ? incoming : [incoming];
+        if (values.length > 0) {
+            const incomingValues: Array<DataOption> = [];
+            values.forEach((v) => {
+                // Map incoming objects to data option values
+                v.id = v.element_id || v.id;
```

Comment:
what is element_id ? I think that was hacked in one of the backbone models, is that still used ?

Reply:
Thanks, I removed it.